test/e2e/framework/util.go

```diff
@@ -0,0 +1,186 @@
+	fieldSelector := fields.OneTermEqualSelector("metadata.name", name).String()
+	lw := &cache.ListWatch{
+		ListFunc: func(options metav1.ListOptions) (runtime.Object, error) {
+			options.FieldSelector = fieldSelector
+			return dynamicClient.Resource(resource).Namespace(namespace).List(context.TODO(), options)
+		},
+		WatchFunc: func(options metav1.ListOptions) (watch.Interface, error) {
+			options.FieldSelector = fieldSelector
+			return dynamicClient.Resource(resource).Namespace(namespace).Watch(context.TODO(), options)
+		},
+	}
+	_, err := watchtools.UntilWithSync(
+		ctx,
+		lw,
+		&unstructured.Unstructured{},
+		func(store cache.Store) (bool, error) {
+			_, exists, err := store.Get(&metav1.ObjectMeta{Namespace: namespace, Name: name})
+			if err != nil {
+				return true, err
+			}
+			if !exists {
+				return true, nil
+			}
+			return false, nil
+		},
+		func(e watch.Event) (bool, error) {
+			switch t := e.Type; t {
+			case watch.Added, watch.Bookmark:
+				return false, nil
+			case watch.Modified:
+				// DeltaFIFO can return modified event on re-list if the object is recreated in the meantime
+				if e.Object.(metav1.Object).GetUID() != uid {
+					return true, nil
+				}
+				return false, nil
+			case watch.Deleted:
+				return true, nil
+			case watch.Error:
+				return true, apierrors.FromObject(e.Object)
+			default:
+				return true, fmt.Errorf("unexpected event type %v", t)
+			}
+		},
+	)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// TODO: Replace when Scylla cluster can configure a SA or use the default one.
+func SetupScyllaClusterSA(ctx context.Context, coreClient corev1client.CoreV1Interface, rbacClient rbacv1client.RbacV1Interface, namespace, scyllaClusterName string) error {
+	sa, err := coreClient.ServiceAccounts(namespace).Create(ctx, &corev1.ServiceAccount{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: fmt.Sprintf("%s-member", scyllaClusterName),
+		},
+	}, metav1.CreateOptions{})
+	if err != nil {
+		return err
+	}
+
+	_, err = rbacClient.RoleBindings(namespace).Create(ctx, &rbacv1.RoleBinding{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: sa.Name,
+		},
+		Subjects: []rbacv1.Subject{
+			{
+				APIGroup:  corev1.GroupName,
+				Kind:      rbacv1.ServiceAccountKind,
+				Namespace: sa.Namespace,
+				Name:      sa.Name,
+			},
+		},
+		RoleRef: rbacv1.RoleRef{
+			APIGroup: rbacv1.GroupName,
+			Kind:     "ClusterRole",
+			Name:     "scylla-manager-cluster-member",
+		},
+	}, metav1.CreateOptions{})
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func DumpEventsInNamespace(ctx context.Context, c kubernetes.Interface, namespace string) {
+	events, err := c.CoreV1().Events(namespace).List(ctx, metav1.ListOptions{})
+	o.Expect(err).NotTo(o.HaveOccurred())
+
+	By(fmt.Sprintf("Found %d events.", len(events.Items)))
+	// Sort events by their first timestamp
+	sortedEvents := events.Items
+	if len(sortedEvents) > 1 {
+		sort.Sort(byFirstTimestamp(sortedEvents))
```

Comment:
nit: you can inline less function, it's short enough.
sort.Slice(sortedEvents, func(){...})

Reply:
I don't really have a preference there, so why not :)